Shared/Objects/FilterDrawerButton.swift

```diff
@@ -0,0 +1,101 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2023 Jellyfin & Jellyfin Contributors
+//
+
+import Defaults
+import Foundation
+
+enum FilterDrawerButton: String, CaseIterable, Defaults.Serializable, Displayable, Identifiable {
+
+    case filters
+    case genres
+    case order
+    case sort
+
+    var displayTitle: String {
+        switch self {
+        case .filters:
+            return L10n.filters
+        case .genres:
+            return L10n.genres
+        case .order:
+            return L10n.order
+        case .sort:
+            return L10n.sort
+        }
+    }
+    
+    var id: String {
+        rawValue
+    }
+
+    var settingsFilter: WritableKeyPath<ItemFilters, [ItemFilters.Filter]> {
+        switch self {
+        case .filters:
+            return \.filters
+        case .genres:
+            return \.genres
+        case .order:
+            return \.sortOrder
+        case .sort:
+            return \.sortBy
+        }
+    }
+    
+    var settingsSelectorType: SelectorType {
+        switch self {
+        case .filters, .genres:
+            return .multi
+        case .order, .sort:
+            return .single
+        }
+    }
+    
+    var settingsItemsFilterProperty: String {
```

Comment:
This is now unused.

Reply:
Ah, yes it is. Removed. I realized that this was the same class name as the one in Swiftfin/Components/FilterDrawerHStack. I've renamed this to FilterDrawerButtonSelection. TBH, I am terrible at naming things and I'm unsure what the convention is for Swift. Let me know if this should be named something else.